internal/rcm/api.go

```diff
@@ -0,0 +1,148 @@
+// Package rcm provides alternative HTTP API to Weldr.
+// It's primary use case is for the RCM team. As such it is driven solely by their requirements.
+package rcm
+
+import (
+	"encoding/json"
+	"log"
+	"net"
+	"net/http"
+
+	"github.com/google/uuid"
+	"github.com/julienschmidt/httprouter"
+	"github.com/osbuild/osbuild-composer/internal/blueprint"
+	"github.com/osbuild/osbuild-composer/internal/store"
+	"github.com/osbuild/osbuild-composer/internal/target"
+)
+
+// API encapsulates RCM-specific API that is exposed over a separate TCP socket
+type API struct {
+	logger *log.Logger
+	store  *store.Store
+	router *httprouter.Router
+}
+
+// New creates new RCM API
+func New(logger *log.Logger, store *store.Store) *API {
+	api := &API{
+		logger: logger,
+		store:  store,
+		router: httprouter.New(),
+	}
+
+	api.router.RedirectTrailingSlash = false
+	api.router.RedirectFixedPath = false
+	api.router.MethodNotAllowed = http.HandlerFunc(methodNotAllowedHandler)
+	api.router.NotFound = http.HandlerFunc(notFoundHandler)
+
+	api.router.POST("/submit", api.submit)
```

Comment:
I recommend using
```
POST /compose
GET /compose/:uuid
```
This should more in line with general REST principles.

Reply:
Sound reasonable to me.

Reply:
Agreed.